### ISSUE TITLE
AppleALC

### ISSUE THREAD

**Issue by idoo** [rafaelmaeuer/Asus-UX32VD-Hackintosh#29]
As I understand `AppleALC` should work with this laptop, but I can't get it to work. Any ideas?

https://github.com/acidanthera/AppleALC/wiki/Supported-codecs

**Comment by rafaelmaeuer** [comment]
Do you have problems with VodooHDA?

**Comment by almontic** [comment]
@idoo I used at first VodooHDA but the switch between speakers and earphones wasn't automatic and it was quite annoying. I tried last version of AppleALC and everything works fine.

**Comment by rafaelmaeuer** [comment]
I tested AppleALC and it works really nice with automatic input switch, so I replaced VoodooHDA in favour of AppleALC.

**Comment by rafaelmaeuer** [comment]
During the rebuild with OpenCore I noticed that there is no support of internal mic using alcid=3. So I tested all ASUS-related layouts found in [ALC269/Info.plist](https://github.com/acidanthera/AppleALC/blob/master/Resources/ALC269/Info.plist) with following results:

- alcid=3: audio + switch, no mic (Asus K53SJ, Asus G73s)
- alcid=12: audio, no switch + mic (Asus Vivobook S200CE)
- alcid=19: audio + switch + mic (Asus Vivobook S300CA)
- alcid=45: audio + switch + mic + Line-In (Asus A45A 269VB1)
- alcid=93: audio + switch, no mic + switch (Asus K53SJ Mod)

So the best results were achieved with layout-ids 19 and 45, but layout 19 shows a lot of unmatched audio-devices. Therefore layout 45 will be used.
